### DISCUSSION TITLE
testing with react-testing-library

### DISCUSSION THREAD

**Discussion by af-ffx** [TanStack/router#198] [Q&A] [answered]
I tried to setup testing with react-location and react-testing-library. It seems like react-location does not match any route. Even wildcard ones are not rendered. Does anyone have a similar problem? Or maybe can point me in the right direction for fixing it?

I prepared a small example to show my problem:
[https://codesandbox.io/s/react-location-testing-library-zjrot?file=/index.test.js](https://codesandbox.io/s/react-location-testing-library-zjrot?file=/index.test.js)

**Comment by tannerlinsley** [comment, marked as answer]
React Locations routing is asynchronous, so you'll need to use tools like `waitFor`: https://codesandbox.io/s/react-location-testing-library-forked-3o8jr?file=/index.test.js

**Reply by af-ffx** [reply]
My bad! Thank you for your help!

**Comment by pauldvu** [comment]
For those running into this issue, my tests were **"passing"** even when they were suppose too fail.  It's an easy fix just  make sure you `add` async await

``` javascript
const mainText = "Home";

describe("App", () => {
  it("It should show home route", async () => {
    const { getByText } = render(<App />);
    await waitFor(() => getByText(mainText));
  });
});
```
